organization/src/test/scala/com/improving/app/organization/domain/OrganizationSpec.scala

```diff
@@ -1,35 +1,366 @@
-      actualEvent.info shouldBe command.info
-      actualEvent.timestamp > 0 shouldBe true
-      testKit.currentState shouldBe
-        OrgState(actualEvent.orgId, actualEvent.info, actualEvent.timestamp)
-      result.reply.orgId shouldBe command.orgId
-      result.reply.info shouldBe command.info
-      result.reply.timestamp shouldBe actualEvent.timestamp
+import org.scalatest.wordspec.AnyWordSpecLike
+
+import scala.util.Random
+
+object OrganizationSpec {
+  val config: Config = ConfigFactory.parseString("""
+    akka.persistence.journal.plugin = "akka.persistence.journal.inmem"
+    akka.persistence.journal.inmem.test-serialization = on
+
+    akka.actor.serialization-bindings{
+      "com.improving.app.common.serialize.PBMsgSerializable" = proto
     }
-  }*/
+  """)
 }
+class OrganizationSpec
+    extends ScalaTestWithActorTestKit(OrganizationSpec.config)
+    with AnyWordSpecLike
+    with BeforeAndAfterAll
+    with Matchers {
+  override def afterAll(): Unit = testKit.shutdownTestKit()
+
+  def createTestVariables(): (String, ActorRef[OrganizationRequestEnvelope], TestProbe[StatusReply[OrganizationEvent]]) = {
+    val organizationId = Random.nextString(31)
+    val p = this.testKit.spawn(Organization(PersistenceId.ofUniqueId(organizationId)))
+    val probe = this.testKit.createTestProbe[StatusReply[OrganizationEvent]]()
+    (organizationId, p, probe)
+  }
+
+  def establishOrganization(
+    organizationId: String,
+    p: ActorRef[OrganizationRequestEnvelope],
+    probe: TestProbe[StatusReply[OrganizationEvent]],
+    organizationInfo: OrganizationInfo = baseOrganizationInfo
+  ): StatusReply[OrganizationEvent] = {
+    p ! Organization.OrganizationRequestEnvelope(
+      EstablishOrganization(
+        organizationId = Some(OrganizationId(organizationId)),
+        onBehalfOf = Some(MemberId("establishingUser")),
+        organizationInfo = Some(organizationInfo)
+      ),
+      probe.ref
+    )
+
+    probe.receiveMessage()
+  }
+
+  "A Organization Actor" when {
+    // Should this also error out when other organizations have the same name?
+    "in the Uninitialized State" when {
+      "executing EstablishOrganization command" should {
+        "error for an unauthorized updating user" ignore {
+          val (organizationId, p, probe) = createTestVariables()
+
+          p ! Organization.OrganizationRequestEnvelope(
+            EstablishOrganization(
+              organizationId = Some(OrganizationId(organizationId)),
+              onBehalfOf = Some(MemberId("unauthorizedUser")),
+              organizationInfo = Some(baseOrganizationInfo)
+            ),
+            probe.ref
+          )
+
+          val response = probe.receiveMessage()
+          assert(response.isError)
+
+          val responseError = response.getError
+          responseError.getMessage shouldEqual "User is not authorized to modify Organization"
+
+        }
+
+        "succeed for the golden path and return the proper response" in {
```

Comment:
have you checked test cases on the validation of the data itself? None on organizationId? etc

Reply:
likewise for the rest of the commands

Reply:
I added some integration testing for bad request handling. Does that satisfy your concern here?